hardware_interface/test/test_robot_hardware_interface.cpp

```diff
@@ -0,0 +1,180 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <string>
+#include <vector>
+
+#include "gmock/gmock.h"
+
+#include "hardware_interface/robot_hardware.hpp"
+#include "hardware_interface/types/hardware_interface_return_values.hpp"
+
+namespace hw = hardware_interface;
+
+namespace
+{
+class MyTestRobotHardware : public hw::RobotHardware
+{
+public:
+  hw::hardware_interface_ret_t init() override
+  {
+    return hw::HW_RET_OK;
+  }
+
+  hw::hardware_interface_ret_t read() override
+  {
+    return hw::HW_RET_OK;
+  }
+
+  hw::hardware_interface_ret_t write() override
+  {
+    return hw::HW_RET_OK;
+  }
+};
+
+const auto JOINT_NAME = "joint_1";
+const auto NEW_JOINT_NAME = "joint_2";
+}  // namespace
+
+namespace testing
+{
+class TestRobotHardwareInterface : public Test
+{
+protected:
+  void SetUp()
+  {
+    robot.init();
+  }
+
+  void SetUpHandles()
+  {
+    robot.register_joint_command_handle(&pos_command_handle);
+    robot.register_joint_state_handle(&state_handle);
+    robot.register_operation_mode_handle(&op_mode_handle);
+  }
+
+  void SetUpNewHandles()
+  {
+    robot.register_joint_command_handle(&new_pos_command_handle);
+    robot.register_joint_state_handle(&new_state_handle);
+    robot.register_operation_mode_handle(&new_op_mode_handle);
+  }
+
+  MyTestRobotHardware robot;
```

Comment:
I prefer member variables to end with an underscore.

Reply:
underscored all of 'em